File: services/user-service/src/main/java/com/visor/school/userservice/model/UserRole.java
package com.visor.school.userservice.model;

/**
 * User roles in the School Management System
 * Role hierarchy: SUPER_ADMIN > ADMINISTRATOR > TEACHER/STUDENT/PARENT
 */
public enum UserRole {
    SUPER_ADMIN,
    ADMINISTRATOR,
    TEACHER,
    STUDENT,
    PARENT
}

File: services/user-service/src/main/kotlin/com/visor/school/userservice/model/UserRole.kt
package com.visor.school.userservice.model

/**
 * User roles in the School Management System
 * Role hierarchy: SUPER_ADMIN > ADMINISTRATOR > TEACHER/STUDENT/PARENT
 */
enum class UserRole {
    SUPER_ADMIN,
    ADMINISTRATOR,
    TEACHER,
    STUDENT,
    PARENT
}

